src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1012,6 +1012,22 @@ public static void onEmptyLeftClick(EntityPlayer player, ItemStack stack)
         MinecraftForge.EVENT_BUS.post(new PlayerInteractEvent.LeftClickEmpty(player, stack));
     }
 
+    public static net.minecraftforge.event.entity.player.CriticalHitEvent getCriticalHit(EntityPlayer player, Entity target, boolean vanillaCritical, float vanillaDefault)
+    {
+        if (!vanillaCritical)
+            vanillaDefault = 1F; //So mods can easily check if vanilla say critical or not. 
+        net.minecraftforge.event.entity.player.CriticalHitEvent hitResult = new net.minecraftforge.event.entity.player.CriticalHitEvent(player, target, vanillaDefault);
+        net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(hitResult);
+        if(hitResult.getResult() != Event.Result.DENY)
+        {
+            if(hitResult.getResult() == Event.Result.ALLOW || vanillaCritical)
```

Comment:
Well result is not so confusing, (in my opinion) as ALLOW is for enabkling even normal hits to be critical, DENY whould cacle der critical hit (if itwas one), and default whould just do the vanilla stuff (so falling but not sprinting not on ladder and so on... )